tarok/src/cards.cpp

```diff
@@ -84,4 +89,31 @@ std::array<Card, 54> BuildDeck() {
   return deck;
 }
 
+std::tuple<std::vector<int>, std::array<std::vector<int>, 3>> DealCards(
+    int seed = 42) {
+  // create vector of card indices
+  std::vector<int> cards(54);
+  std::iota(std::begin(cards), std::end(cards), 0);
+
+  // shuffle indices
+  auto rng = std::default_random_engine(seed);
+  std::shuffle(std::begin(cards), std::end(cards), rng);
+
+  // first six cards are talon
+  std::vector<int>::iterator start(cards.begin());
+  std::vector<int>::iterator end(cards.begin() + 6);
+  std::vector<int> talon(start, end);
+
+  std::array<std::vector<int>, 3> private_cards{};
+  // every player gets next 16 cards
+  std::advance(start, 6);
+  for (int i = 0; i < 3; i++) {
```

Comment:
In order to avoid complex refactoring when we introduce the fourth player, I would add the `num_players_` member to the game and use it throughout the codebase. In fact, I think it might make sense to make code general enough where possible so that we don't have to change too much when adding another player (e.g. change the fact that everyone gets 16 cards).